src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.jsx

```diff
@@ -126,6 +126,19 @@ const GroupDetails = createReactClass({
           Sentry.withScope(scope => {
             Sentry.captureException(new Error('Project not found'));
           });
+        } else {
+          const locationWithProject = {...this.props.location};
+          if (
+            locationWithProject.query.project === undefined &&
+            locationWithProject.query._allp === undefined
+          ) {
+            //We use _allp as a temporary measure to know they came from the issue list page with no project selected (all projects included in filter).
+            //If it is not defined, we add the locked project id to the URL (this is becayse uf someone navigates directly to an issue on single-project priveledges, then goes back - they were getting assigned to the first project).
+            //If it is defined, we do not so that our back button will bring us to the issue list page with no project selected instead of the locked proejct.
```

Comment:
```suggestion
            //If it is defined, we do not so that our back button will bring us to the issue list page with no project selected instead of the locked project.
```